### PR TITLE
mito-ai: replace first match

### PR DESCRIPTION
**Pull request by aarondr77**
# Description

When the streamlit agent finds multiple matches, defaults to replacing the first one. 

# Testing

See the new test

# Documentation

No

---

> [!NOTE]
> Change search/replace to replace only the first occurrence and log when multiple matches are found; update tests accordingly.
> 
> - **Streamlit conversion**:
>   - `apply_search_replace` in `mito_ai/streamlit_conversion/search_replace_utils.py` now:
>     - Logs multiple matches via `log("mito_ai_search_text_found_multiple_times", ...)` instead of raising.
>     - Replaces only the first occurrence using `result.replace(search_text, replace_text, 1)`.
> - **Tests**:
>   - Add test to verify only the first occurrence is replaced when duplicates exist.
>   - Remove test expecting an error on multiple matches.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 4ffcee43d49f695b3085ba0bd9dc9ede726de0f5. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: mito-ai/mito_ai/tests/streamlit_conversion/test_apply_search_replace.py [hunk 1]
```diff
@@ -200,6 +200,23 @@
 
 st.title("🚀 My App")
 st.write("Welcome to my application")"""
+    ),
+    
+    # Test case 11: Only replace first occurrence when search text exists multiple times
```
Comment (PoojasreeDwarakanath) [inline mito-ai/mito_ai/tests/streamlit_conversion/test_apply_search_replace.py:205]: I think we can leave the replace instruction to the agent as is. If that fails to include surrounding lines, then we can replace the first occurence.

**Review by PoojasreeDwarakanath — APPROVED** [review]
Looks good to me
